debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java

```diff
@@ -1118,6 +1119,110 @@ public void shouldAllowForExportedSnapshot() throws Exception {
         VerifyRecord.isValidInsert(s2recs.get(0), PK_FIELD, 3);
     }
 
+    @Test
+    @FixFor("DBZ-2288")
+    @SkipWhenDecoderPluginNameIs(value = SkipWhenDecoderPluginNameIs.DecoderPluginName.PGOUTPUT, reason = "PgOutput needs publication for manually created slot")
+    public void exportedSnapshotShouldNotSkipRecordOfParallelTx() throws Exception {
+        TestHelper.dropDefaultReplicationSlot();
+        TestHelper.createDefaultReplicationSlot();
+
+        // Testing.Print.enable();
+        TestHelper.execute(SETUP_TABLES_STMT);
+        TestHelper.execute(INSERT_STMT);
+
+        Configuration config = TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.SNAPSHOT_MODE, SnapshotMode.EXPORTED.getValue())
+                .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, Boolean.FALSE)
+                .with(PostgresConnectorConfig.MAX_QUEUE_SIZE, 2)
+                .with(PostgresConnectorConfig.MAX_BATCH_SIZE, 1)
+                .build();
+        final PostgresConnection pgConnection = TestHelper.create();
+        pgConnection.setAutoCommit(false);
+        pgConnection.executeWithoutCommitting(INSERT_STMT);
+        final AtomicBoolean inserted = new AtomicBoolean();
+        start(PostgresConnector.class, config, loggingCompletion(), x -> false, x -> {
+            if (!inserted.get()) {
+                TestHelper.execute(INSERT_STMT);
+                try {
+                    pgConnection.commit();
+                }
+                catch (Exception e) {
+                    e.printStackTrace();
+                }
+                inserted.set(true);
+            }
+        });
+        assertConnectorIsRunning();
+
+        // Consume records from the snapshot
+        SourceRecords actualRecords = consumeRecordsByTopic(4);
+
+        // Consume records from concurrent transactions
+        actualRecords = consumeRecordsByTopic(4);
+
+        List<SourceRecord> s1recs = actualRecords.recordsForTopic(topicName("s1.a"));
+        List<SourceRecord> s2recs = actualRecords.recordsForTopic(topicName("s1.a"));
+        s2recs = actualRecords.recordsForTopic(topicName("s2.a"));
+        assertThat(s1recs.size()).isEqualTo(2);
+        assertThat(s2recs.size()).isEqualTo(2);
+
+        stopConnector();
+        TestHelper.dropDefaultReplicationSlot();
+    }
+
+    @Test
+    @FixFor("DBZ-2288")
+    @SkipWhenDecoderPluginNameIsNot(value = SkipWhenDecoderPluginNameIsNot.DecoderPluginName.PGOUTPUT, reason = "Publication not supported")
+    public void exportedSnapshotShouldNotSkipRecordOfParallelTxPgoutput() throws Exception {
```

Comment:
Couldn't you just create the publication always, and it only would be used when running the test for pgoutput? That way we'd keep a single test.

Reply:
No, as postgres 9 dows not support publications. So it is either based on postgres version or plugin used.

Reply:
Gotcha, makes sense.